File: src/net/eduard/api/lib/old/Tag.java
package net.eduard.api.lib.old;

/**
 * Tag do jogador<br>
 * Versão anterior {@link TagSetup} 1.0
 * 
 * @version 1.0
 * @since 0.7
 * @author Eduard
 * @deprecated Versão Atual {@link net.eduard.api.lib.game.Tag}
 *
 */

public class Tag {
	private String prefix;
	private String suffix;

	public Tag(String prefix, String suffix) {
		setPrefix(prefix);
		setSuffix(suffix);
	}

	public String getPrefix() {
		return this.prefix;
	}

	public String getSuffix() {
		return this.suffix;
	}

	public void setPrefix(String prefix) {
		this.prefix = Principal.getText(prefix);
	}

	public void setSuffix(String suffix) {
		this.suffix = Principal.getText(suffix);
	}
}

File: src/main/java/net/eduard/api/lib/game/Tag.kt
package net.eduard.api.lib.game

import net.eduard.api.lib.modules.Copyable


/**
 * Representa uma Tag geralmente usado para Cargos de Permissão
 * @author Eduard
 * @version 2.0
 * @since EduardAPI 1.0
 */
class Tag(
        var name: String,
        var prefix: String,
        var suffix: String,
        var rank: Int
) {
    constructor() : this("Tag", "","",0)
    constructor(  name: String,
                  prefix: String,
                  rank: Int) : this(name, prefix,"",rank)
    constructor(name: String,
                  prefix: String) : this(name, prefix,"",0)

    fun copy(): Tag {
        return Copyable.copyObject(this)
    }

    override fun toString(): String {
        return "Tag [prefix=$prefix, suffix=$suffix, name=$name, rank=$rank]"
    }
}